File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/IntradayBarRequestElementString.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/IntradayBarRequestElementString.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Element;
import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Schema;

import java.lang.StringBuilder;

public class IntradayBarRequestElementString extends Element
{
    private final String _elementName, _value;

    public IntradayBarRequestElementString(String elementName, String value)
    {
        this._elementName = elementName;
        this._value = value;
    }
    
    public Name name() throws Exception
    {
    	return new Name(this._elementName);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.STRING;
    }
    
    protected String getSecurity()
    {
    	return this._value;
    }
    
    public Object getValue()
    {
        return this._value;
    }
    
    public String getValueAsString()
    {
        return this._value;
    }
    
    public String getValueAsString(int index) throws Exception
    {
        if (index == 0)
            return this._value;
        else
            return super.getValueAsString(index);
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
        String tabs = com.bloomberglp.blpapi.IndentType.Indent(tabIndent);
        StringBuilder result = new StringBuilder();

        result.append(String.format("%s%s = %s%s", tabs, this._elementName, this._value, System.getProperty("line.separator")));

        return result;
    }

}

File: BEmu_cpp/src/IntradayBarRequest/IntradayBarRequestElementString.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayBarRequest/IntradayBarRequestElementString.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "BloombergTypes/Name.h"
#include "BloombergTypes/ElementPtr.h"
#include "IntradayBarRequest/IntradayBarRequestElementString.h"

namespace BEmu
{
	namespace IntradayBarRequest
	{
		IntradayBarRequestElementString::IntradayBarRequestElementString(const std::string& elementName, const std::string& value, bool surroundWithQuotes) : 
			CanConvertToStringType(value, surroundWithQuotes)
		{
			this->_elementName = elementName;
			this->_value = value;
		}

		IntradayBarRequestElementString::~IntradayBarRequestElementString()
		{
		}

		Name IntradayBarRequestElementString::name() const
		{
			Name result(this->_elementName.c_str());
			return result;
		}

		const std::string& IntradayBarRequestElementString::security() const
		{
			return this->_value;
		}

		size_t IntradayBarRequestElementString::numValues() const
		{
			return 1;
		}

		size_t IntradayBarRequestElementString::numElements() const
		{
			return 0;
		}

		bool IntradayBarRequestElementString::isArray() const
		{
			return false;
		}

		bool IntradayBarRequestElementString::isComplexType() const
		{
			return false;
		}

		const char * IntradayBarRequestElementString::getValueAsString(int index) const
		{
			if(index == 0)
				return this->_value.c_str();
			else
				throw elementPtrEx;
		}

		std::ostream& IntradayBarRequestElementString::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->ValueAsString());
			return stream;
		}
	}
}